### ISSUE TITLE
JioSaavn

### ISSUE THREAD

**Issue by skyme5** [yt-dlp/yt-dlp#2486]
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.01.21**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

India

### Example URLs

- Song: https://www.jiosaavn.com/song/ik-mili-mainu-apsraa/ByFacyZYGgs
- Album: [https://www.jiosaavn.com/album/pyaar-karte-ho-na/qjDZ,EIA4Ic_](https://www.jiosaavn.com/album/pyaar-karte-ho-na/qjDZ,EIA4Ic_)
- Album: [https://www.jiosaavn.com/album/dil-galti-kar-baitha-hai-feat.-mouni-roy/EDxgblL9zAA_](https://www.jiosaavn.com/album/dil-galti-kar-baitha-hai-feat.-mouni-roy/EDxgblL9zAA_)
- Playlist: [https://www.jiosaavn.com/featured/shaan-sings/EW64eF0QIjY_](https://www.jiosaavn.com/featured/shaan-sings/EW64eF0QIjY_)
- Podcast: [https://www.jiosaavn.com/shows/playlist/1/g4LaiQkXmok_](https://www.jiosaavn.com/shows/playlist/1/g4LaiQkXmok_)
- Artist: [https://www.jiosaavn.com/artist/javed-ali-/iJXgfejIuJQ_](https://www.jiosaavn.com/artist/javed-ali-/iJXgfejIuJQ_)

### Description

Audio streaming website in India. I'm willing to write the implementation.

### Verbose log

```shell
[debug] Command-line config: ['-vU', 'https://www.jiosaavn.com/song/ik-mili-mainu-apsraa/ByFacyZYGgs']
[debug] Encodings: locale cp1252, fs utf-8, out cp1252 (No ANSI), err utf-8, pref cp1252
[debug] yt-dlp version 2022.01.21 [f20d607b0]
[debug] Python version 3.10.2 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg N-105336-g7247a6fed8-20220121 (setts), ffprobe N-105336-g7247a6fed8-20220121
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.jiosaavn.com/song/ik-mili-mainu-apsraa/ByFacyZYGgs
WARNING: [generic] Falling back on generic information extractor.
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.jiosaavn.com/song/ik-mili-mainu-apsraa/ByFacyZYGgs
Traceback (most recent call last):
  File "C:\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1367, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1437, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\common.py", line 612, in extract
    ie_result = self._real_extract(url)
  File "C:\Python\Python310\lib\site-packages\yt_dlp\extractor\generic.py", line 3966, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.jiosaavn.com/song/ik-mili-mainu-apsraa/ByFacyZYGgs
```

**Comment by Ashish0804** [comment]
>I'm willing to write the implementation

Just make PR straight then ig
But not worth it tbh.
Worse quality than YT, last i checked
Also: https://github.com/deepjyoti30/ytmdl/blob/master/ytmdl/meta/saavn.py

**Comment by skyme5** [comment]
@Ashish0804 thanks. I think for my case [ytmdl](https://github.com/deepjyoti30/ytmdl) maybe enough.

**Comment by Ashish0804** [comment]
FYI: ytmdl downloads the music from YouTube and then tags the track with savaan metadata, afaik. 
I linked it so you can get a little bit of help in finding the endpoints. You are still free to make a PR (and maybe reopen this issue)
@skyme5

**Comment by skyme5** [comment]
@Ashish0804 sure, nothing urgent but will add this to my TODO list.

**Comment by nitin2953** [comment]
JioSaavn is not working.

The site just fetch mp4/m4a files,
I can download them with help of browser's devtools.

### Verbose log
```
[debug] Command-line config: ['-vU', 'https://www.jiosaavn.com/song/masakali/BF80RTVdfUU']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.11.1 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 1.1.1q  5 Jul 2022)
[debug] exe versions: ffmpeg 5.1.2-full_build-www.gyan.dev (setts), ffprobe 5.1.2-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.2
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[generic] Extracting URL: https://www.jiosaavn.com/song/masakali/BF80RTVdfUU
[generic] BF80RTVdfUU: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] BF80RTVdfUU: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.jiosaavn.com/song/masakali/BF80RTVdfUU
Traceback (most recent call last):
  File "C:\Users\nitin\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nitin\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\YoutubeDL.py", line 1594, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "C:\Users\nitin\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\common.py", line 694, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nitin\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\generic.py", line 2510, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.jiosaavn.com/song/masakali/BF80RTVdfUU
```

**Comment by pukkandan** [comment]
The issue was closed by OP, presumably because they no longer wanted the feature. Consider opening a new issue if you do.
